engineering/documentation.md

```diff
@@ -0,0 +1,226 @@
+everywhere in the gender spectrum, different races, and different sensibilities.
+I do not wish to alienate any of them, so I need to be careful with the words I choose.
+
+All of these concrete examples can be boiled down to the following best practices.
+
+### State a clear goal
+
+A piece of documentation has a clearly stated goal and audience.
+
+For instance, a `README.md` should be an overview of the project, maybe
+including some basic usage instructions. On the other hand `CONTRIBUTING.md`
+should cover what contributors care about.
+
+Do not make a first time visitor read through how to contribute a project before
+they understand what the project is about.
+
+### Define a scope
+
+As Carl Sagan [famously said](https://www.youtube.com/watch?v=7s664NsLeFM)
+"If you wish to make an apple pie from scratch, you must first invent the universe".
+Would you start an apple pie recipe with quantum physics? Hopefully not.
+
+Same goes for documentation. You need to define the scope of concepts that deserve
+to be explained in a document. Those considered either too advanced or as prerequisites
+should be ommited. You can always link to some other document explaining them.
+
+For instance, if a project uses Go and gRPC feel free to link to the homepages
+of those projects. Do not use more than one short sentence to explain what Go
+or gRPC are.
+
+### Keep it simple
+
+Short sentences win. Avoid having overly complex sentences.
+It might seem weird at first. You might find the resulting text immature.
+But keeping your sentences short help the readers, and that should trump everything else.
+
+Following the same logic, avoid overly complicated words.
+
+As a rule of thumb, sentences longer than a printed line are probably too long.
+
+### Don't alienate the reader
+
+Avoid gendered pronouns. "He or she" should be rewritten as "they".
+There is no reason for your document to make assumptions on the reader's identity.
+
+Avoid sexist connotations.
+For instance, "mom-proof" implies mothers are not technical. Tell that to Grace Hopper.
+You can read more about sexist language and how to avoid it
+[here](https://www.hamilton.edu/academics/centers/writing/writing-resources/avoiding-sexist-language).
+
+Avoid ableist language. Avoid words like "crazy" or "insane".
+More on ableist language [here](https://www.copyediting.com/disabling-ableist-language/#.Wk6yeFQ-dE4).
+
+In general, if someone would find what you wrote offensive it's time to change it.
+There's no excuse to offend anyone in a technical document.
+
+### Take accessibility in mind
+
+It is probably impossible to write documentation that is fully accessible to every single person in the world.
+This doesn't mean we should not strive for documentation that is as accessible as possible.
+
+Many readers might have different degrees of visual impairments.
+While using images avoid attaching meaning to colors that are traditionally hard to distinguish.
```

Comment:
@campoy There are images that are complex to describe with an **alt** attribute, like some data visualisations. We can aim - maybe @dpordomingo can help how to do this in markdown - to make complex images as usable and accessible as possible. There are some [techniques to improve accessibility on complex images](https://www.w3.org/WAI/tutorials/images/complex/)